File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Services/Configuration/ConfigurationService.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Konfigurator_aspnetcore_Backend.Models;
using Konfigurator_aspnetcore_Backend.Models.DTOs;
using Konfigurator_aspnetcore_Backend.Data;
using Microsoft.EntityFrameworkCore;

namespace Konfigurator_aspnetcore_Backend.Services.Configuration
{
	public class ConfigurationService : IConfigurationService
	{
		private readonly LabyrinthKonfiguratorContext _context;

		public ConfigurationService(LabyrinthKonfiguratorContext context)
		{
			_context = context;
		}

		public async Task<ServiceResponseDto<GetConfigurationDataResponseDto>> GetConfigurationData()
		{
            // Response mit den fuer den gesamten Konfigurationsprozess notwendigen Daten erstellen

			// Daten aus der Datenbank auslesen
            var barrierDesigns = await _context.barrierDesigns.ToListAsync();
			var characterDesigns = await _context.characterDesigns.ToListAsync();
			var targetDesigns = await _context.targetDesigns.ToListAsync();
			var tileDesigns = await _context.tileDesigns.ToListAsync();

			var barrierDesignDtos = new List<DesignDto>();
			var characterDesignDtos = new List<DesignDto>();
			var targetDesignDtos = new List<DesignDto>();
			var tileDesignDtos = new List<DesignDto>();

			// Daten mappen
			foreach(var barrierDesign in barrierDesigns)
			{
				barrierDesignDtos.Add(new DesignDto(barrierDesign.barrier, barrierDesign.img));
			}

            foreach(var characterDesign in characterDesigns)
			{
				characterDesignDtos.Add(new DesignDto (characterDesign.character, characterDesign.img));
			}

			foreach(var targetDesign in  targetDesigns)
			{
				targetDesignDtos.Add(new DesignDto (targetDesign.target, targetDesign.img));
			}

			foreach(var tileDesign in tileDesigns)
			{
				tileDesignDtos.Add(new DesignDto (tileDesign.tile, tileDesign.img));
			}

			var response = new GetConfigurationDataResponseDto(
				barrierDesignDtos,
				characterDesignDtos,
				targetDesignDtos,
				tileDesignDtos);

            return new ServiceResponseDto<GetConfigurationDataResponseDto> { Data = response, Success = true };
		}

		public async Task<ServiceResponseDto<AddGameMapResponseDto>> CreateGame(AddGameMapRequestDto request)
		{
			/*
			 * Daten mappen
			 */

			// GameMap
			var gameMap = new GameMap
			{
				name = request.name,
				username = request.username,
				sizeX = request.size.x,
				sizeY = request.size.y,
				tile = request.tile,
				barrier = request.barrier,
				character = request.character,
				target = request.target,
				startPosX = request.startPos.x,
				startPosY = request.startPos.y,
				endPosX = request.endPos.x,
				endPosY = request.endPos.y
			};

			await _context.gameMaps.AddAsync(gameMap);
			await _context.SaveChangesAsync();

			int assigendGameMapId = gameMap.gameMapId;

			// ObstaclesInMap
			var obstaclesInMap = new List<ObstaclesInMap>();

			foreach (var obstacle in request.obstacles)
			{
				obstaclesInMap.Add(new ObstaclesInMap
				{
					gameMapId = assigendGameMapId,
					x = obstacle.x,
					y = obstacle.y
				});
			}

			foreach (var entry in obstaclesInMap)
			{
				await _context.AddAsync(entry);
			}
			await _context.SaveChangesAsync();

			// GameMapId der neu erstellen Map zurueckgeben
			AddGameMapResponseDto response = new AddGameMapResponseDto(assigendGameMapId);

			return new ServiceResponseDto<AddGameMapResponseDto> { Data = response, Success = true, Message = "Game gespeichert" };
		}
	}
}

File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/configuration/configuration.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm'
import { Repository } from 'typeorm'
import { BarrierDesigns } from 'src/entities/BarrierDesignsEntity';
import { CharacterDesigns } from 'src/entities/CharacterDesignsEntity';
import { TargetDesigns } from 'src/entities/TargetDesignsEntity';
import { TileDesigns } from 'src/entities/TileDesignsEntity';
import { GetConfigurationDataResponseDto } from '../dto/get-ConfigurationDataResponse.dto'
import { ServiceResponseDto } from 'src/dto/ServiceResponse.dto';
import { AddGameMapRequestDto } from 'src/dto/add-GameMapRequest.dto';
import { GameMaps } from 'src/entities/GameMapsEntity';
import { ObstaclesInMaps } from 'src/entities/ObstaclesInMapsEntity';
import { AddGameMapResponseDto } from 'src/dto/add-GameMapResponse.dto';
import { DesignsDto } from 'src/dto/Designs.dto';

@Injectable()
export class ConfigurationService {
  constructor(
    @InjectRepository(BarrierDesigns) private barrierDesignsRepository: Repository<BarrierDesigns>,
    @InjectRepository(CharacterDesigns) private characterDesignsRepositroy: Repository<CharacterDesigns>,
    @InjectRepository(TargetDesigns) private targetDesignsRepositroy: Repository<TargetDesigns>,
    @InjectRepository(TileDesigns) private tileDesignsRepositroy: Repository<TileDesigns>,
    @InjectRepository(GameMaps) private gameMapsRepositroy: Repository<GameMaps>,
    @InjectRepository(ObstaclesInMaps) private obstaclesInMapsRepositroy: Repository<ObstaclesInMaps>
  ) {}

  // Response mit den fuer den gesamten Konfigurationsprozess notwendigen Daten erstellen
  async GetConfigurationData() {
    // Daten aus der Datenbank auslesen
    let barrierDesigns = await this.barrierDesignsRepository.find();
    let characterDesigns = await this.characterDesignsRepositroy.find();
    let targetDesigns = await this.targetDesignsRepositroy.find();
    let tileDesigns = await this.tileDesignsRepositroy.find();

    let barrierDesignDtos: DesignsDto[] = [];
    let characterDesignDtos: DesignsDto[] = [];
    let targetDesignDtos: DesignsDto[] = [];
    let tileDesignDtos: DesignsDto[] = [];

    /*
     * Daten mappen
     */

    barrierDesigns.forEach(element => {
      let designTemp = new DesignsDto(element.barrier, element.img.toString('base64'));

      barrierDesignDtos.push(designTemp);
    });

    characterDesigns.forEach(element => {
      let designTemp = new DesignsDto(element.character, element.img.toString('base64'));

      characterDesignDtos.push(designTemp);
    });

    targetDesigns.forEach(element => {
      let designTemp = new DesignsDto(element.target, element.img.toString('base64'));

      targetDesignDtos.push(designTemp);
    });

    tileDesigns.forEach(element => {
      let designTemp = new DesignsDto(element.tile, element.img.toString('base64'));

      tileDesignDtos.push(designTemp);
    });

    let response = new GetConfigurationDataResponseDto(
      barrierDesignDtos,
      characterDesignDtos,
      targetDesignDtos,
      tileDesignDtos);

    return new ServiceResponseDto<GetConfigurationDataResponseDto>(response, true, "Daten abgerufen");
  }


  async CreateGame(request: AddGameMapRequestDto) {
    const gameMap = new GameMaps();
    gameMap.name = request.name;
    gameMap.username = request.username;
    gameMap.sizeX = request.size.x;
    gameMap.sizeY = request.size.y;
    gameMap.tile = request.tile;
    gameMap.barrier = request.barrier;
    gameMap.character = request.character;
    gameMap.target = request.target;
    gameMap.startPosX = request.startPos.x;
    gameMap.startPosY = request.startPos.y;
    gameMap.endPosX = request.endPos.x;
    gameMap.endPosY = request.endPos.y;

    const assignedGameMap = await this.gameMapsRepositroy.save(gameMap);

    // ObstaclesInMap
    let obstaclesInMaps: ObstaclesInMaps[] = [];

    request.obstacles.forEach(element => {
      let obstaclesInMapsTemp = new ObstaclesInMaps;
      obstaclesInMapsTemp.gameMapId = assignedGameMap.gameMapId;
      obstaclesInMapsTemp.x = element.x;
      obstaclesInMapsTemp.y = element.y;

      obstaclesInMaps.push(obstaclesInMapsTemp);
    })

    obstaclesInMaps.forEach(async element => {
      await this.obstaclesInMapsRepositroy.save(element);
    })

    // GameMapId der neu erstellen Map zurueckgeben
    const response = new AddGameMapResponseDto(assignedGameMap.gameMapId);

    return new ServiceResponseDto<AddGameMapResponseDto>(response, true, "Game gespeichert");
  }
}
